src/v/iceberg/schema.cc

```diff
@@ -105,9 +106,14 @@ void schema::assign_fresh_ids() {
         to_visit_stack.pop_back();
         if (!field) {
             continue;
+        } else if (field->is_add()) {
+            field->id = next_id++;
         }
+        vassert(
+          field->id != 0,
+          "Nested field '{}' wasn't assigned an ID",
+          field->name);
```

Comment:
This feels like it should always be true, but is there a guarantee in code that this never fires? Could a buggy catalog cause a crash, for example?

Reply:
yeah, i've gone back and forth on this. I think the assertion holds for structs processed through record translation, but...

>could a buggy catalog cause a crash

good point, hadn't considered that. I think it's fair to say that a field w/ a zero ID here is strictly ill formed, but since that value could have come from disk probably a non-fatal error would be more appropriate.